### PR TITLE
hwmv2: Port renesas socs/boards

### PR DESCRIPTION
**Pull request by nordicjm**

### CODE REVIEW

File: soc/renesas/rcar/rcar_gen3/CMakeLists.txt
Comment (xakep-amatop) [inline soc/renesas/rcar/rcar_gen3/CMakeLists.txt:12]: note: these files will be moved in scoope of this pr https://github.com/zephyrproject-rtos/zephyr/pull/67357

File: soc/renesas/rcar/rcar_gen3/CMakeLists.txt
Comment (xakep-amatop) [inline soc/renesas/rcar/rcar_gen3/CMakeLists.txt:1]: add EPAM Systems copyright too

File: soc/renesas/rcar/rcar_gen3/Kconfig
Comment (xakep-amatop) [inline soc/renesas/rcar/rcar_gen3/Kconfig:16]: as far as I see, you've renamed this SoC family

File: soc/renesas/rcar/rcar_gen3/Kconfig.defconfig
Comment (xakep-amatop) [inline soc/renesas/rcar/rcar_gen3/Kconfig.defconfig:10]: you may align the number of IRQs with R7, actually it is the same according to documentation 480 SPI and 32 PPI+SGI

File: soc/renesas/rcar/rcar_gen4/Kconfig.soc
Comment (xakep-amatop) [inline soc/renesas/rcar/rcar_gen4/Kconfig.soc:8]: all series have a specific Cortex R core, look like we need to avoid it

File: soc/renesas/rzt2m/CMakeLists.txt
Comment (xakep-amatop) [inline soc/renesas/rzt2m/CMakeLists.txt:1]: I believe that the dir architecture should be another for the RZ, like `soc/renesas/rz/t` or `soc/renesas/rz/a`, moreover it will be aligned with rz dts dir (https://github.com/zephyrproject-rtos/zephyr/tree/main/dts/arm/renesas/rz)

look at renesas doc page: https://www.renesas.com/us/en/products/microcontrollers-microprocessors/rz-mpus

File: boards/renesas/rcar_spider_s4/rcar_spider_s4.yaml
Comment (xakep-amatop) [reply boards/renesas/rcar_spider_s4/rcar_spider_s4.yaml:2]: I meant to say that you've connected the Spider S4 board to the Cortex R52 in name.

**Comment by xakep-amatop** [comment]
Could you please add appropriate changes to the MAINTAINER file to Renesas sections?

File: boards/renesas/rcar_spider_s4/rcar_spider_s4.yaml
Comment (nordicjm) [reply boards/renesas/rcar_spider_s4/rcar_spider_s4.yaml:2]: The yaml will need to be updated when the other core is added to have the CPU identifier so will become `rcar_spider_s4_<cpu>.yaml` and identifier will match that

File: soc/renesas/rcar/rcar_gen3/CMakeLists.txt
Comment (nordicjm) [reply soc/renesas/rcar/rcar_gen3/CMakeLists.txt:12]: That PR will need to be rebased on this as hwmv2 will be merged after 3.6 has been complete prior to anything else and only doc/bug fixes are accepted for 3.6 now

File: soc/renesas/rzt2m/CMakeLists.txt
Comment (nordicjm) [reply soc/renesas/rzt2m/CMakeLists.txt:1]: This is how it is named in the current zephyr tree, if you would like it changed that is fine but I request that you submit a PR for it after this is merged, there's about 640 boards in total to convert in a very short time frame and I don't have time available to make changes for vendors to each of them

**Comment by nordicjm** [comment]
> Could you please add appropriate changes to the MAINTAINER file to Renesas sections?

This will be done *en masse* once all boards and socs have been ported in the branch

File: soc/renesas/rcar/rcar_gen3/Kconfig
Comment (nordicjm) [reply soc/renesas/rcar/rcar_gen3/Kconfig]: Removed

File: soc/renesas/rcar/rcar_gen3/CMakeLists.txt
Comment (nordicjm) [reply soc/renesas/rcar/rcar_gen3/CMakeLists.txt:1]: Added

**Review by soburi — APPROVED** [review]
LGTM about RA part.

**Comment by carlescufi** [comment]
@xakep-amatop please take another look

File: boards/renesas/rcar_salvator_xs_m3/board.yml
Comment (firscity) [inline boards/renesas/rcar_salvator_xs_m3/board.yml:2]: Actually M3 is another name for r8a7796 SoC family. IIUC meaning of hwmv2, then generic board should be called `rcar_salvator_xs` and `m3` should be one of the SoC options.

File: boards/arduino/arduino_uno_r4_minima/Kconfig.arduino_uno_r4_minima
Comment (firscity) [inline boards/arduino/arduino_uno_r4_minima/Kconfig.arduino_uno_r4_minima:5]: Is this change actually related to Renesas platform?

File: boards/renesas/rcar_h3_salvatorx/board.yml
Comment (firscity) [inline boards/renesas/rcar_h3_salvatorx/board.yml:2]: (see comment about M3 Salvator-XS board)
H3 is also another name for r8a7795 SoC, so board should be `rcar_salvator_x`

File: snippets/xen_dom0/snippet.yml
Comment (nordicjm) [reply snippets/xen_dom0/snippet.yml:10]: Snippet support is current not working in hwmv2

File: boards/arduino/arduino_uno_r4_minima/Kconfig.arduino_uno_r4_minima
Comment (nordicjm) [reply boards/arduino/arduino_uno_r4_minima/Kconfig.arduino_uno_r4_minima:5]: It's a renesas chip, so yes?

File: boards/renesas/rcar_h3_salvatorx/board.yml
Comment (nordicjm) [reply boards/renesas/rcar_h3_salvatorx/board.yml]: Fixed

File: boards/renesas/rcar_salvator_xs_m3/board.yml
Comment (nordicjm) [reply boards/renesas/rcar_salvator_xs_m3/board.yml]: Fixed

File: boards/renesas/rcar_salvator_xs/doc/index.rst
Comment (firscity) [inline boards/renesas/rcar_salvator_xs/doc/index.rst:50]: Spurious change, I guess)

Comment (nordicjm) [reply boards/renesas/rcar_salvator_xs/doc/index.rst]: Indeed, accidental placeholder

File: snippets/xen_dom0/snippet.yml
Comment (firscity) [inline snippets/xen_dom0/snippet.yml:13]: This part should be also changed to rcar_salvator_xs/r8a77961 (same for tests yaml's)

File: snippets/xen_dom0/snippet.yml
Comment (firscity) [reply snippets/xen_dom0/snippet.yml:10]: I think it worth to at least mention in commit message or generate error if snippet is used for HWMv2 board, I spent some time yesterday to understand why it is not building properly. In current PR version it will impact 2 boards (salvator_xs and h3ulcb).

File: boards/renesas/rcar_h3ulcb/doc/rcar_h3ulcb_r7.rst
Comment (firscity) [inline boards/renesas/rcar_h3ulcb/doc/rcar_h3ulcb_r7.rst:212]: `boards/renesas/rcar_h3ulcb/doc/rcar_h3ulcb_a57.rst` should have same changes

**Review by firscity — CHANGES_REQUESTED** [review]

**Review by gmarull — DISMISSED** [review]
update docs/maintainers as needed

File: snippets/xen_dom0/snippet.yml
Comment (nordicjm) [reply snippets/xen_dom0/snippet.yml:10]: It is tracked in the main hwmv2 issue here: https://github.com/zephyrproject-rtos/zephyr/issues/51831

File: boards/renesas/rcar_h3ulcb/doc/rcar_h3ulcb_r7.rst
Comment (nordicjm) [reply boards/renesas/rcar_h3ulcb/doc/rcar_h3ulcb_r7.rst:212]: Updated

File: snippets/xen_dom0/snippet.yml
Comment (nordicjm) [reply snippets/xen_dom0/snippet.yml]: Updated

**Review by xakep-amatop — COMMENTED** [review]
Tested changes on H3ULCB CA57 and Salvator XS with disk performance/synchronization test/sample. Everything works as expected.

File: snippets/xen_dom0/snippet.yml
Comment (nordicjm) [reply snippets/xen_dom0/snippet.yml]: Fixed

File: boards/renesas/rcar_salvator_xs/rcar_salvator_xs.dts
Comment (nordicjm) [reply boards/renesas/rcar_salvator_xs/rcar_salvator_xs.dts]: Fixed

File: boards/renesas/rcar_salvator_xs/doc/index.rst
Comment (nordicjm) [reply boards/renesas/rcar_salvator_xs/doc/index.rst:34]: Fixed

File: soc/renesas/rcar/rcar_gen3/Kconfig.soc
Comment (nordicjm) [reply soc/renesas/rcar/rcar_gen3/Kconfig.soc]: Fixed

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: boards/renesas/rcar_h3ulcb/Kconfig.rcar_h3ulcb
Comment (nordicjm) [reply boards/renesas/rcar_h3ulcb/Kconfig.rcar_h3ulcb]: Fixed

**Review by tejlmand — APPROVED** [review]

File: snippets/xen_dom0/boards/rcar_salvator_xs3.overlay
Comment (xakep-amatop) [inline snippets/xen_dom0/boards/rcar_salvator_xs3.overlay:1]: You forgot to delete `3` at the end of the file name.

File: boards/renesas/rcar_salvator_xs/rcav_salvator_xs-pinctrl.dtsi
Comment (xakep-amatop) [inline boards/renesas/rcar_salvator_xs/rcav_salvator_xs-pinctrl.dtsi:1]: please, fix name too `rcav_...` --> `rcar_...`

File: snippets/xen_dom0/boards/rcar_salvator_xs3.overlay
Comment (nordicjm) [reply snippets/xen_dom0/boards/rcar_salvator_xs3.overlay:1]: Fixed

File: boards/renesas/rcar_salvator_xs/rcav_salvator_xs-pinctrl.dtsi
Comment (nordicjm) [reply boards/renesas/rcar_salvator_xs/rcav_salvator_xs-pinctrl.dtsi:1]: Fixed

**Review by xakep-amatop — APPROVED** [review]

**Comment by nordicjm** [comment]
@firscity @gmarull please re-review

**Review by firscity — APPROVED** [review]

**Pull request merged**
